src/query/src/sql.rs

```diff
@@ -182,6 +183,40 @@ pub async fn show_tables(
                 .context(error::CreateRecordBatchSnafu)?;
             Ok(Output::RecordBatches(records))
         }
+        ShowKind::Full => {
+            let mut table_types = Vec::new();
+            for table_name in &tables {
+                let table_type = catalog_manager
+                    .table(query_ctx.current_catalog(), &schema_name, table_name)
+                    .await
+                    .context(error::CatalogSnafu)?
+                    .unwrap()
+                    .table_type();
+
+                let table_type = match table_type {
+                    table::metadata::TableType::Base => "BASE TABLE",
```

Comment:
Maybe we can implement `Display` trait for `TableType`, then simply this code into:

```rust
let table_type = format!("{table_type}");
```

https://doc.rust-lang.org/std/fmt/trait.Display.html

Reply:
Yeah, I actually did that at first but I was getting some weird errors with the compiler saying `TableType` isn't a value.
I'll try it again and drop a screenshot

Reply:
I found my mistake